deployment/src/main/java/org/amqphub/quarkus/qpid/jms/deployment/QpidJmsProcessor.java

```diff
@@ -80,6 +91,25 @@ public void enableSecurityServices(BuildProducer<FeatureBuildItem> feature,
     AdditionalBeanBuildItem registerBean() {
         return AdditionalBeanBuildItem.unremovableOf(QpidJmsProducer.class);
     }
+    
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void connectionFactoryWrapper(Optional<ConnectionFactoryWrapperBuildItem> connectionFactoryWrapper,
+                                  QpidJmsRecorder recorder,
+                                  BuildProducer<SyntheticBeanBuildItem> syntheticBeanProducer) {
+        Optional<Function<ConnectionFactory, Object>> wrapper = Optional.empty();
+        if (connectionFactoryWrapper.isPresent()) {
+            wrapper = Optional.of(connectionFactoryWrapper.get().getWrapper());
+        }
+        
+        SyntheticBeanBuildItem.ExtendedBeanConfigurator configurator = SyntheticBeanBuildItem.configure(ConnectionFactoryWrapper.class)
+                .setRuntimeInit()
+                .defaultBean()
+                .scope(ApplicationScoped.class)
+                .runtimeValue(recorder.getConnectionFactoryWrapper(wrapper));
+        
+       syntheticBeanProducer.produce(configurator.done());
+    }
```

Comment:
If I understand this the idea is that if someone got the quarkus-pooled-jms extension on their classpath explicitly or indirectly (or really any other code similarly creating such a ConnectionFactoryWrapperBuildItem) then with this change as it stands, quarkus-qpid-jms would simply execute the wrapper function provided and let the other extension/code control what actually gets 'returned' by quarkus-qpid-jms for its created ConnectionFactory...with the use case here being such that quarkus-pooled-jms for example would create a JmsPoolConnectionFactory and just start pooling.

I dont think think quarkus-qpid-jms extension should just start doing pooling (or whatever another similar function happened to do). I think it should be opt-in via config toggle on the client extension side so it must be chosen over and above just another dependency becoming present. Using the pool can very significantly change the behaviour of using the client, far beyond just not creating new 'real' connections whenever the application does (or more likely framework with tendency to), they should have to choose it.

I'm not really sure what a nice name for that config toggle would be. The change is being done around enabling pooling specifically, but the mechanism doesnt seem in any way fixed to that, so its not clear an obviously pooling-specific named config actually makes complete sense vs something more general.

(I mostly actually think the components that themselves essentially require pooling, should be the thing supplying it rather than requiring many individual clients that could be used with it to do so)

Reply:
Thanks for taking review these changes!

> If I understand this the idea is that if someone got the quarkus-pooled-jms extension on their classpath explicitly or indirectly (or really any other code similarly creating such a ConnectionFactoryWrapperBuildItem) then with this change as it stands, quarkus-qpid-jms would simply execute the wrapper function provided and let the other extension/code control what actually gets 'returned' by quarkus-qpid-jms for its created ConnectionFactory...with the use case here being such that quarkus-pooled-jms for example would create a JmsPoolConnectionFactory and just start pooling.

Yeah, you are right. Anyway there is a config in `quarkus-pooled-jms` to enable/disable the pooling.
```
quarkus.pooled-jms.pooling.enabled = true | false
```

So it might be a little bit confusion if we introduce a config toggle in `quarkus-qpid-jms`. Also I think if users include `quarkus-pooled-jms` in their projects or even create `ConnectionFactoryWrapperBuildItem`, they should know what they want actually.

> (I mostly actually think the components that themselves essentially require pooling, should be the thing supplying it rather than requiring many individual clients that could be used with it to do so)

Yeah, I agree and the most suitable way is to use [Decorator](https://quarkus.io/guides/cdi#decorators) but unfortunately it does not work with method producer beans, see the disscussion [here](https://groups.google.com/g/quarkus-dev/c/thdTCj-XNUU).

Reply:
I know about the quarkus.pooled-jms.pooling.enabled setting, it isnt really sufficient, its too late by that point. Plus anything could add that dependency, the user wouldnt necessarily 100% know it is there. I also wouldnt necessarily know it pooling when someone asks for help and says they are using this extension. The change in behaviour is significant, it should be opt-in on this side. As comparison, Spring has support for this same pool in its starters, it is opt-in in both config terms and in adding the dependency.

Plus again, since ConnectionFactoryWrapperBuildItem is not actually specific to pooling, anything else could decide to use it. I think this extension should by default control what it creates/returns, which it simply wouldnt if enabling this by default.

Reply:
That makes sense and yeah `quarkus-pooled-jms` provides XA integration as well. So the hardest thing is to choose the name for this opt-in, how about `quarkus.qpid-jms.wrap.enabled` with default `false`?

Reply:
I think that works...simple, general, and reflective of what actually happens.